pkg/setting/setting.go

```diff
@@ -373,6 +373,7 @@ func applyEnvVariableOverrides(file *ini.File) error {
 
 func envKey(sectionName string, keyName string) string {
 	sN := strings.ToUpper(strings.Replace(sectionName, ".", "_", -1))
+	sN = strings.Replace(sN, "-", "_", -1)
 	kN := strings.ToUpper(strings.Replace(keyName, ".", "_", -1))
```

Comment:
Can the `keyName` contain also hyphens? If yes then they should also be replaced.

Reply:
In theory yes, but we always use a-z and _ in current Grafana settings so lets skip that for now.